### DISCUSSION TITLE
Get typescript types for custom screens (or other custom theme object)

### DISCUSSION THREAD

**Discussion by alexandersokolow** [tailwindlabs/tailwindcss#10349] [Help]
I have a hook that takes a tailwind breakpoint and checks whether the screen height corresponds to that breakpoint.
I want the hook to only be able to take actual tailwind breakpoints as specified in the config (for example, I also have a breakpoint called "3xl").

I could define a type `"xs" | "sm" | ... | "2xl" | "3xl"` but that would not be very nice, because I would have to ensure that the type in this file corresponds to my config.
It would be much better if I could automatically generate this type from the typescript config and then use it in my hook.

To achieve this I tried: 
1. importing the screens object from the typescript config (into a variable called screens) and using `keyof typeof screens`. Unfortunately, that didn't work, it gave me the never type.
2. adding an extra typescript-file, creating the screen object there, then importing it into both the tailwind config and the hook (where I could then easily use the keyof thingy to get the type). Unfortunately, the tailwind config won't recognize the typescript file, I guess due to how webpack / transpiling works. Even when I try to make it a js-file and import that, it causes issues with the postcss-processor & I'm not sure if I could extract the types from the js-file anyways.

Is there some neat way of solving this problem?

**Comment by uigywnkiub** [comment]
This is works for me
```js
import defaultTheme from 'tailwindcss/defaultTheme'

export const getBreakpointWidth = (
  breakpoint: keyof typeof defaultTheme.screens,
) => {
  return console.log(breakpoint)
}
```
